src/extensions/askUserForParam.ts

```diff
@@ -116,6 +129,81 @@ export async function askForParam(paramSpec: Param): Promise<string> {
   return response;
 }
 
+async function promptSecret(
+  projectId: string,
+  instanceId: string,
+  paramSpec: Param
+): Promise<string> {
```

Comment:
Can you check whether this works works correctly in the following case? I have a non-secret param (API_KEY), and I'm updating to a new version where API_KEY is now a secret.

I think this function will assume that a Secret already exists and break

Reply:
This work -- tested. I made sure that "type" change counts as param change. The comparator compares both param name and type.